Android/src/org/droidplanner/android/fragments/geotag/GeoTagImagesService.java

```diff
@@ -0,0 +1,389 @@
+    public static final String STATE_FINISHED_GEOTAGGING = PACKAGE_NAME + ".FINISHED_GEOTAGGING";
+    public static final String STATE_PROGRESS_UPDATE_GEOTAGGING = PACKAGE_NAME + ".PROGRESS_UPDATE_GEOTAGGING";
+
+    public static final String EXTRA_SUCCESS = PACKAGE_NAME + ".EXTRA_SUCCESS";
+    public static final String EXTRA_FAILURE_MESSAGE = PACKAGE_NAME + ".EXTRA_FAILURE_MESSAGE";
+    public static final String EXTRA_PROGRESS = PACKAGE_NAME + ".EXTRA_PROGRESS";
+    public static final String EXTRA_TOTAL = PACKAGE_NAME + ".EXTRA_TOTAL";
+    public static final String EXTRA_GEOTAGGED_FILES = PACKAGE_NAME + ".EXTRA_GEOTAGGED_FILES";
+
+    private AsyncTask getTlogsTask;
+    private GeoTagTask geoTagTask;
+    private LocalBroadcastManager lbm;
+
+    @Override
+    public void onCreate() {
+        super.onCreate();
+
+        lbm = LocalBroadcastManager.getInstance(getApplicationContext());
+    }
+
+    @Override
+    public int onStartCommand(Intent intent, int flags, int startId) {
+        if (intent == null) {
+            return super.onStartCommand(intent, flags, startId);
+        }
+
+        switch (intent.getAction()) {
+            case ACTION_START_LOADING_LOGS:
+                startLoadingLogs();
+                break;
+            case ACTION_CANCEL_LOADING_LOGS:
+                cancelLoadingLogs();
+                break;
+            case ACTION_START_GEOTAGGING:
+                startGeoTagging();
+                break;
+            case ACTION_CANCEL_GEOTAGGING:
+                cancelGeoTagging();
+                break;
+        }
+        return super.onStartCommand(intent, flags, startId);
+    }
+
+    @Override
+    public IBinder onBind(Intent intent) {
+        return null;
+    }
+
+    private void startLoadingLogs() {
+        if (getTlogsTask != null) {
+            getTlogsTask.cancel(true);
+        }
+        getTlogsTask = new DownloadCameraTlogs(getApplicationContext());
+        getTlogsTask.execute();
+    }
+
+    private void cancelLoadingLogs() {
+        if (getTlogsTask != null) {
+            getTlogsTask.cancel(true);
+            getTlogsTask = null;
+        }
+    }
+
+    private void startGeoTagging() {
+        File folder = getApplicationContext().getExternalFilesDir(null);
+        File tlogFile = new File(folder.getPath() + "/camera_msgs.tlog");
+        Timber.d("path: " + tlogFile.getPath());
+
+        geoTagImages(getApplicationContext(), tlogFile);
+    }
+
+    private void cancelGeoTagging() {
+        if (geoTagTask != null) {
+            geoTagTask.cancel(true);
+            geoTagTask = null;
+        }
+    }
+
+    private void geoTagImages(final Context context, File tlogFile) {
+        final String extMount = getExternalStorage(context);
+        if (extMount == null) {
+            return;
+        }
+
+        final ArrayList<File> photoFiles = new ArrayList<>();
+
+        List<File> photos = searchDir(extMount);
```

Comment:
@ne0fhyk I think it was left over from before where I was looking through multiple directories. I'll remove it